tls/s2n_ktls.h

```diff
@@ -53,6 +53,8 @@ int s2n_ktls_record_writev(struct s2n_connection *conn, uint8_t content_type,
 /* These functions will be part of the public API. */
 int s2n_connection_ktls_enable_send(struct s2n_connection *conn);
 int s2n_connection_ktls_enable_recv(struct s2n_connection *conn);
+int s2n_sendfile(struct s2n_connection *conn, int in_fd, off_t offset, size_t count,
+        size_t *bytes_written, s2n_blocked_status *blocked);
```

Comment:
I had that conversation with Cameron :) I think the conclusion was that removing the blocked argument from new APIs would sacrifice future flexibility with no real gain to usability.
* This can only block on write _for now_, but because the other send APIs include `blocked` we might use it for something more useful in the future. Like, for example, a callback blocking. I don't practically see that happening, but removing it from one API removes the option from all APIs. Then send really will only ever block on write.
* Even if we don't use `blocked` here, all the other send APIs still use it and so the application can still misuse all the other send APIs. We don't really solve the larger problem by preventing misuse for the most niche send API.
So basically... we're including `blocked` to match all the other send APIs.